### DISCUSSION TITLE
Pydeck: visualization without jupyter

### DISCUSSION THREAD

**Discussion by iegorval** [visgl/deck.gl#9019] [Q&A] [answered]
Hi all, it seems I am missing some important piece of information. All examples & articles on PyDeck demonstrate its usage in Jupyter notebook. I need to run it without Jupyter by just creating a window. Is this available directly through PyDeck (just running `.show()` on deck that worked in Jupyter does not do the trick)? If yes, how? If no, what would you recommend me for the best view of the html-rendered deck?

**Comment by donmccurdy** [comment, marked as answer]
Outside of Jupyter (and certain other notebook environments) pydeck [generates an HTML file](https://github.com/visgl/deck.gl/blob/f2564dad79909a815aa2235c79da8edd763c806e/bindings/pydeck/examples/geojson_layer.py#L39), but does not:

1. run a local server to host that HTML page, or
2. launch a window in any particular browser to view the locally-hosted HTML

For steps (1) you could [use any Python local HTTP server](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/Tools_and_setup/set_up_a_local_testing_server#using_python), or similar Node.js servers, or desktop apps like [WorldWideWeb](https://apps.apple.com/us/app/worldwideweb-desktop/id1621370168?mt=12). Some of these local servers might open a browser window for you, others may require you to copy/paste the URL into the browser yourself.

**Comment by iegorval** [comment]
Thanks for a fast answer! Was not sure if I should make a local HTTP or if there is some functionality directly in PyDeck which I cannot find.
